.mci.yml

```diff
@@ -971,6 +971,31 @@ buildvariants:
         - name: clang-tidy
         - name: compile_without_tests
 
+    - name: ubuntu1404-debug-gcc
+      display_name: "Ubuntu 14.04 Debug (g++ 4.8)"
+      expansions:
+        build_type: "Debug"
+        tar_options: *linux_tar_options
+        cmake_flags: *linux_cmake_flags
+        mongodb_version: *version_latest
+        compiler: g++
+      run_on:
+        - ubuntu1404-small
+      tasks:
+        - name: compile_without_tests
+
+    - name: ubuntu1404-debug-clang
+      display_name: "Ubuntu 14.04 Debug (clang++ 3.4)"
+      expansions:
+        build_type: "Debug"
+        tar_options: *linux_tar_options
+        cmake_flags: *linux_cmake_flags
+        mongodb_version: *version_latest
+        compiler: clang++
+      run_on:
+        - ubuntu1404-small
+      tasks:
+        - name: compile_without_tests
```

Comment:
I have added clang 3.4 and gcc 4.8 to the matrix.

Travis was testing clang 3.6 and gcc 4.9, older than anything we were testing. So I added this to ensure we don't lose coverage against older compilers.

This is using strictly less versions, but I am making an assumption that a successful build on clang 3.4 / gcc 4.8 will imply a successful build on clang 3.6 / gcc 4.9.